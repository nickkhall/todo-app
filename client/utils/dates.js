export const getPrettyDate = (timestamp) => {
  if (timestamp) {
    let date = new Date(timestamp * 1000).toString();
    date = date.substring(0, date.indexOf('GMT') - 1);

    return date;
  }
  return 'Unknown Date';
};
