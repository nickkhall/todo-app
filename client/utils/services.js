/* eslint-disable no-console */
import axios from 'axios';

// Errors
import { API_ERRORS } from 'copy/Global/errors';

export const defaultCatch = (endPoint, source = 'services') => (e) => {
  console.error(`${source}/${endPoint}`, e);
  const errMessage = e.response.data.message || e.response.data.error;
  throw new Error(errMessage, e);
};
/**
 * Request wrapper for all request types (GET, POST, PUT, DELETE, etc).
 * @param {!Object} config The request configuration.
 * @param {string} config.url The URL for the request.
 * @param {string} config.method The method (GET, POST, etc) for the request.
 * @param {!Object} config.payload Optional request payload.
 * @return {!Promise} Promise containing the axios response.
 */
export const makeRequest = (url, method, payload) => axios({
  method,
  url,
  ...(payload ? { data: payload } : {})
})
  .then((res) => {
    if (res && (res.status >= 200 || res.status < 400)) {
      return res;
    }
    console.log('make request was successful');
    throw new Error(API_ERRORS.BAD_REQUEST);
  })
  .then((res) => {
    if ('location' in res.headers) {
      return makeRequest(res.headers.location, 'GET', null, true);
    }

    return res;
  });
