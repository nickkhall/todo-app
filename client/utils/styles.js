export const combineStyles = (...styles) => function CombineStyles(theme) {
  const outStyles = styles.map((arg) => {
    // Apply the "theme" object for style functions.
    if (typeof arg === 'function') {
      return arg(theme);
    }
    // Objects need no change.
    return arg;
  });

  return outStyles.reduce((acc, val) => Object.assign(acc, val));
};

export const hexToRGB = (hex, alpha = 1) => {
  let parseString = hex;
  if (hex.startsWith('#')) { parseString = hex.slice(1, 7); }
  if (parseString.length !== 6) { return null; }
  const r = parseInt(parseString.slice(0, 2), 16);
  const g = parseInt(parseString.slice(2, 4), 16);
  const b = parseInt(parseString.slice(4, 6), 16);
  if (Number.isNaN(r) || Number.isNaN(g) || Number.isNaN(b)) { return null; }
  return `rgba(${r}, ${g}, ${b}, ${alpha})`;
};
