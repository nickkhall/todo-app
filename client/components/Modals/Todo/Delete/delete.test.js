import { testCommonComponentAttrs } from 'utils/tests';

// Components
import DeleteTodoModal from './delete';

const mockProps = {
  name: 'foobar'
};

describe('<DeleteTodoModal />', () => {
  testCommonComponentAttrs(DeleteTodoModal, mockProps);
});
