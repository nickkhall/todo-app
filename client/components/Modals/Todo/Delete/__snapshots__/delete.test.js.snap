// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteTodoModal /> when rendering should render a component matching the snapshot 1`] = `""`;
